src/models/streaming/BufferedDuplex.ts

```diff
@@ -17,37 +16,35 @@ export class BufferedDuplex extends Duplex {
         super(options);
         this.input = input;
         this.output = output;
-        this.areOutputHandlersSetup = false;
+        this.setupInput();
```

Comment:
These handlers were the main thing we were missing before. They ensure that events from the inner streams are propagated to users of our BufferedDuplex